rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -94,6 +94,17 @@ bool Player::is_storage_completely_loaded() const
 
 void Player::play(const PlayOptions & options)
 {
+  if (reader_->has_next()) {
+    // Reader does not have "peek", so we must "pop" the first message to see its timestamp
+    auto message = reader_->read_next();
+    prepare_clock(options, message->time_stamp);
+    // Make sure that first message gets played by putting it into the play queue
+    message_queue_.enqueue(message);
```

Comment:
isn't this kind of contradicting with `load_storage_content`? Why are we explicitly load the first (and only the first) message in the queue?

Reply:
In order to know its timestamp - we were already doing this, I just moved it to a slightly different place.

Reply:
load_storage_content picks up at the next message and keeps loading - we've called this before starting that process so it has a strict ordering